editor/plugins/material_editor_plugin.cpp

```diff
@@ -164,9 +164,22 @@ void MaterialEditor::_button_pressed(Node *p_button) {
 MaterialEditor::MaterialEditor() {
 	// canvas item
 
+	vc_2d = memnew(SubViewportContainer);
+	vc_2d->set_stretch(true);
+	add_child(vc_2d);
+	vc_2d->set_anchors_and_offsets_preset(PRESET_FULL_RECT);
+
+	viewport_2d = memnew(SubViewport);
+	Ref<World3D> world_2d;
+	world_2d.instantiate();
+	viewport_2d->set_world_3d(world_2d); //use own world
```

Comment:
Since you aren't rendering any 3D objects, you don't need to create a unique world 3D. By default Viewports will only render 2D objects that are their children.

Reply:
Thanks! I was not aware of that. I modified my commit accordingly. :)